### PR TITLE
Fix spellclick

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- Tested
- Arch Linux

## How to Test the Changes:

1. See video without PR
2. https://user-images.githubusercontent.com/14080516/135752124-d2da5222-da8a-4087-b0ae-77b8f5e6d990.mp4
3. Download https://github.com/IntelligentQuantum/WoWAddin
4. Compile
5. Run WoWAddin.exe

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Handlers/SpellHandler.cpp [hunk 1]
```diff
@@ -646,6 +646,11 @@ void WorldSession::HandleSpellClick(WorldPacket& recvData)
     if (!unit)
         return;
 
+    if (_player->IsHostileTo(unit))
```
Comment (Nefertumm) [inline src/server/game/Handlers/SpellHandler.cpp:649]: Is there any gob/npc hostile to player with spell click? Just to be sure we are not breaking them.
Anyway, nice catch. 😁

**Review by Nefertumm — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by IntelligentQuantum** [comment]
> what's the status of this PR?

Need someone to test it.

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers 

> > what's the status of this PR?
> 
> Need someone to test it.

@IntelligentQuantum this is a small change so if you tested it yourself and are sure it doesn't cause any regression, it can be good to go 👍

**Comment by IntelligentQuantum** [comment]
@FrancescoBorzi Everything works fine as far as I tested.

**Comment by FrancescoBorzi** [comment]
@IntelligentQuantum the codestyle check is failing, please have a look at it

**Comment by Nefertumm** [comment]
All good!

**Comment by Footman** [comment]
breaks https://wotlkdb.com/?npc=38248

sister svalna spear click script

**Comment by Meramaf** [comment]
> breaks https://wotlkdb.com/?npc=38248
> 
> sister svalna spear click script

Confirmed

**Comment by Nyeriah** [comment]
@IntelligentQuantum could you please take a look into this?

**Comment by FrancescoBorzi** [comment]
> @IntelligentQuantum could you please take a look into this?

@IntelligentQuantum  why no answer?

**Comment by IntelligentQuantum** [comment]
@FrancescoBorzi Sorry, I do not check notifications much

@Nyeriah i will check it.

**Pull request merged**